pinot-controller/src/main/java/org/apache/pinot/controller/LeadControllerManager.java

```diff
@@ -216,8 +216,6 @@ synchronized void onResourceConfigChange() {
       leadControllerResourceEnabled = LeadControllerUtils.isLeadControllerResourceEnabled(_helixManager);
     } catch (Exception e) {
       LOGGER.error("Exception when checking whether lead controller resource is enabled or not.", e);
-      _isLeadControllerResourceEnabled = false;
```

Comment:
Does this mean that when we get ResourceConfigChanged notification, if zookeeper is disconnected, we will miss the notification altogether? Or, when we get reconnected, we will re-read the resource configs (we should do this).

Otherwise, if we throw the exception, does helix call us back with the resource config again?

Reply:
So the idea here is remain the state if any exception happened. 
Enabling the resource is always one-off. If administrator wants to enable it he will check the log. 
Plus, I don't think it's rare to have resource config changes because every time there's a Helix task generated, the task will be written to resource config, which will trigger this notification as well.

Reply:
Can you add a comment to that effect? thanks.

Reply:
other than that, lgtm